pytorch_lightning/strategies/fully_sharded_native.py

```diff
@@ -0,0 +1,266 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import contextlib
+import logging
+import os
+from typing import Any, Dict, Generator, List, Optional, Union
+
+import torch
+from torch.distributed.distributed_c10d import _get_default_group, ProcessGroup
+
+import pytorch_lightning as pl
+from pytorch_lightning.overrides.distributed import prepare_for_backward
+from pytorch_lightning.plugins.environments.cluster_environment import ClusterEnvironment
+from pytorch_lightning.plugins.io.checkpoint_plugin import CheckpointIO
+from pytorch_lightning.plugins.precision import PrecisionPlugin
+from pytorch_lightning.strategies.parallel import ParallelStrategy
+from pytorch_lightning.strategies.strategy import TBroadcast
+from pytorch_lightning.trainer.states import TrainerFn
+from pytorch_lightning.utilities import rank_zero_only
+from pytorch_lightning.utilities.distributed import (
+    _get_process_group_backend_from_env,
+    distributed_available,
+    get_default_process_group_backend_for_device,
+)
+from pytorch_lightning.utilities.distributed import group as _group
+from pytorch_lightning.utilities.distributed import init_dist_connection, ReduceOp, sync_ddp_if_available
+from pytorch_lightning.utilities.exceptions import MisconfigurationException
+from pytorch_lightning.utilities.imports import _TORCH_GREATER_EQUAL_1_11
+from pytorch_lightning.utilities.optimizer import optimizers_to_device
+from pytorch_lightning.utilities.seed import reset_seed
+
+if _TORCH_GREATER_EQUAL_1_11:
+    from torch.distributed.fsdp.fully_sharded_data_parallel import (
+        BackwardPrefetch,
+        CPUOffload,
+        FullyShardedDataParallel,
+    )
+    from torch.distributed.fsdp.wrap import enable_wrap
+
+
+log = logging.getLogger(__name__)
+
+
+class DDPFullyShardedNativeStrategy(ParallelStrategy):
```

Comment:
it's inherited from `ParallelStrategy`. Doesn't it launch scripts like DDP? Can't find a launcher configured for it.

Reply:
Its assuming the process launcher is externally configured. Is script launcher required? If so, I can drop in a fn similar to DDP.py

Reply:
I think yes. `DDPSharded` strategy is inherited from DDP and DDP configures the script launcher. So I guess native FSDP should too?

Reply:
No, we should not keep extending the DDP strategy class: https://github.com/PyTorchLightning/pytorch-lightning/issues/11863

It is better to implement exactly what's needed by implementing the ParallelStrategy, and not worry about changes to DDP possibly breaking this subclass.

Reply:
I'm going to propose another followup issue to add in the launcher scripts. I think we can refactor some code out of DDP to share that logic with this plugin.